### ISSUE TITLE
Create API endpoint for index.ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#5]
Create the API endpoint projects/index.ts that will API route for getting all projects. Be sure to add the proper extension when creating the file. Here are the instructions:

## Background

We need to create a new Next.js API endpoint that allows us to fetch all projects for a specific user. This endpoint will be used by the dashboard page to display a list of all the user's projects.

## Task

Create a new API endpoint at `src/pages/api/projects/index.ts` that will handle the `GET` request to fetch all projects for the current authenticated user.

### Requirements

1. Import the necessary dependencies:
   - `import { NextApiRequest, NextApiResponse } from 'next';`
   - `import { getSession } from 'next-auth/client';`
   - `import { supabase } from '../../../lib/supabaseClient';`
   - `import { Project, ProjectSchema } from '../../../types';`

2. Create a new async function called `handler` with the following signature:
   - `async function handler(req: NextApiRequest, res: NextApiResponse): Promise<void>`

3. Inside the `handler` function, check if the request method is `GET`. If not, return a 405 (Method Not Allowed) status code with a JSON response containing an error message:
   - `{ error: 'Method not allowed' }`

4. Retrieve the user session using the `getSession` function from `next-auth/client`. If there's no session, return a 401 (Unauthorized) status code with a JSON response containing an error message:
   - `{ error: 'Unauthorized' }`

5. Use the `supabase` client to fetch all projects from the `projects` table where the `user_id` matches the `id` of the current authenticated user. Make sure to select only the necessary columns from the `projects` table (id, name, description, user_id, created_at, updated_at).

6. Validate the fetched data using the `ProjectSchema` from the `types.ts` file. You can use the `array()` method from Zod to create an array schema:
   - `const projectsArraySchema = ProjectSchema.array();`
   - `const result = projectsArraySchema.safeParse(fetchedData);`

7. If the validation fails, return a 500 (Internal Server Error) status code with a JSON response containing an error message:
   - `{ error: 'Failed to validate fetched projects' }`

8. If the validation is successful, return a 200 (OK) status code with a JSON response containing the array of validated projects.

### Example TypeScript Type for the Output

The output should be an array of `Project` objects as defined in the `types.ts` file:

```typescript
type ProjectOutput = Project[];
```

### Error Cases

- If the request method is not `GET`, return a 405 (Method Not Allowed) status code with a JSON response containing an error message: `{ error: 'Method not allowed' }`
- If the user is not authenticated, return a 401 (Unauthorized) status code with a JSON response containing an error message: `{ error: 'Unauthorized' }`
- If the validation of the fetched data fails, return a 500 (Internal Server Error) status code with a JSON response containing an error message: `{ error: 'Failed to validate fetched projects' }`

### Summary

In summary, you need to create a new Next.js API endpoint at `src/pages/api/projects/index.ts` that handles a `GET` request to fetch all projects for the current authenticated user. Make sure to validate the fetched data using Zod and handle any error cases appropriately.
